src/management-system-v2/app/(dashboard)/[environmentId]/engines/engines-modal.tsx

```diff
@@ -0,0 +1,60 @@
+'use client';
+
+import { Form, Input, Modal } from 'antd';
+import { useEffect } from 'react';
+
+const EnginesModal = ({
+  open,
+  close,
+  title,
+  initialData,
+}: {
+  open: boolean;
+  close: (data?: { address: string; ownName: string }) => void;
+  title: string;
+  initialData?: { address: string; ownName: string };
+}) => {
+  const [form] = Form.useForm();
+
+  useEffect(() => {
```

Comment:
You could just pass the `initialData` to the form, and add `destroyOnClose` to the modal, this way whenever the modal is opened with new data a new Form is mounted that has the correct initial values